### PR TITLE
DataFrame transform: use simple code generation when LLM is not GPT-4

### PR DESCRIPTION
**Pull request by gengliangwang**
Currently, the SDK is using a ReAct agent for DataFrame transform, which requires the output format to be of the following format:
sql

```

Thought: Spark SQL does not support dynamic pivot operations, which are required to transpose the table as requested. I should get all the distinct values of column product.
Action: query_sql_db
Action Input: "SELECT DISTINCT product FROM spark_ai_temp_view_096b0f"
Observation: [('Normal',), ('Mini',), ('Foldable',), ('Pro',), ('Pro Max',)]
```

This doesn't work well when the LLM is not GPT-4. From time to time, there can be parse exceptions. For example: https://github.com/pyspark-ai/pyspark-ai/issues/118

So, to make sure the API `df.ai.transform` generate robust code on non-gpt-4 models, we can follow https://github.com/pyspark-ai/pyspark-ai/pull/159 and implement SQL generator which supports auto-retry on errors:
image

### CODE REVIEW

File: pyspark_ai/spark_sql_chain.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+from langchain.chains import LLMChain
+from langchain.chat_models.base import BaseChatModel
+from langchain.schema import HumanMessage, BaseMessage
+from pyspark.sql import SparkSession
+from pyspark_ai.code_logger import CodeLogger
+
+from pyspark_ai.ai_utils import AIUtils
+
+
+class SparkSQLChain(LLMChain):
```
Comment (gengliangwang) [inline pyspark_ai/spark_sql_chain.py:13]: TODO: we can abstract the code of SparkSQLChain and PythonExecutor to a new common super class.

Comment (gengliangwang) [reply pyspark_ai/spark_sql_chain.py:13]: Caching is the major difference here. I will unify them in a follow-up PR.

File: pyspark_ai/prompt.py [hunk 2]
```diff
@@ -205,6 +216,22 @@
     prefix=SPARK_SQL_PREFIX,
 )
 
+SQL_CHAIN_EXAMPLES = [
+    sql_question1 + f"\nAnswer:\n```{sql_answer1}```",
+    sql_question2 + f"\nAnswer:\n```{sql_answer2}```",
```
Comment (asl3) [inline pyspark_ai/prompt.py:221]: do we want to give it all 4 examples?

**Review by asl3 — APPROVED** [review]

File: pyspark_ai/prompt.py (hunk 2 above)
Comment (gengliangwang) [reply pyspark_ai/prompt.py:221]: I am thinking about removing example 4 in all the agent/chains.

**Pull request merged**